components/pages/ProfileCommunity.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import {
+  View, TextInput, StyleSheet
+} from 'react-native';
+import { Button, useTheme } from 'react-native-paper';
+import ProfileCommunityList from '../lists/ProfileCommunityList';
+
+const sampleData = [
+  {
+    name: 'Best Community',
+    banner: 'https://static.wikia.nocookie.net/spongebobgalaxy/images/2/23/Squidwards_House.png/revision/latest?cb=20180707172828',
+    tag: 'BestTag',
+    favorite: true,
+  },
+  {
+    name: 'Second Best Community',
+    banner: 'https://static.wikia.nocookie.net/spongebobgalaxy/images/2/23/Squidwards_House.png/revision/latest?cb=20180707172828',
+    tag: 'SecondBestTag',
+    favorite: false,
+  },
+];
+
+function ProfileCommunity() {
+  const { colors } = useTheme();
+  const [searchQuery, setSearchQuery] = useState('');
+  const [communityList, setCommunityList] = useState([]);
+  const [filteredCommunityList, setFilteredCommunityList] = useState([]);
+
+  useEffect(() => {
+    // get request to fetch friends list
+    setCommunityList(sampleData);
+    setFilteredCommunityList(sampleData);
+  }, []);
+
+  const handleSearch = (query) => {
+    // need to make the live search to wait(to be delayed a bit)
```

Comment:
if you're just running a filter on an array of pre-loaded communities, like you currently have, I don't actually know if debouncing is necessary. It will seem snappier without it. But if you're going to re-pull data from the DB every time there's a search update, then yes please debounce so we don't hit rate limits :)